TaskLayer/MbrAnalysis/MbrAnalysisRunner.cs

```diff
@@ -0,0 +1,273 @@
+
+            ConcurrentBag<MbrSpectralMatch> bestMbrMatches = new();
+
+            foreach (SpectraFileInfo spectraFile in spectraFiles)
+            {
+                List<ChromatographicPeak> fileSpecificMbrPeaks =
+                    parameters.FlashLfqResults.Peaks[spectraFile].Where(p => p.IsMbrPeak).ToList();
+                if (fileSpecificMbrPeaks == null || (!fileSpecificMbrPeaks.Any())) break;
+
+                MyFileManager myFileManager = new(true);
+                MsDataFile myMsDataFile =
+                    myFileManager.LoadFile(spectraFile.FullFilePathWithExtension, commonParameters);
+                MassDiffAcceptor massDiffAcceptor = SearchTask.GetMassDiffAcceptor(
+                    commonParameters.PrecursorMassTolerance,
+                    parameters.SearchParameters.MassDiffAcceptorType, parameters.SearchParameters.CustomMdac);
+                Ms2ScanWithSpecificMass[] arrayOfMs2ScansSortedByRT = MetaMorpheusTask
+                    .GetMs2Scans(myMsDataFile, spectraFile.FullFilePathWithExtension, commonParameters)
+                    .OrderBy(b => b.RetentionTime).ToArray();
+                string spectralLibraryPath = Path.Combine(parameters.OutputFolder, @"spectralLibrary.msp");
+                SpectralLibrary library = new(new List<string>() { spectralLibraryPath });
+
+                MiniClassicSearchEngine mcse = new(
+                    arrayOfMs2ScansSortedByRT,
+                    massDiffAcceptor,
+                    commonParameters,
+                    library);
+
+                Parallel.ForEach(threads, (i) =>
+                {
+                    // Stop loop if canceled
+                    if (GlobalVariables.StopLoops) { return; }
+
+                    for (; i < fileSpecificMbrPeaks.Count; i += maxThreadsPerFile)
+                    {
+                        ChromatographicPeak mbrPeak = fileSpecificMbrPeaks[i];
+                        PeptideSpectralMatch bestDonorPsm = allPeptides.Where(p =>
+                            p.FullSequence == mbrPeak.Identifications.First().ModifiedSequence).First();
+                        if (bestDonorPsm == null) break;
+                        PeptideWithSetModifications bestDonorPwsm = bestDonorPsm.BestMatchingPeptides.First().Peptide;
+
+                        IEnumerable<PeptideSpectralMatch> peptideSpectralMatches =
+                            mcse.SearchAroundPeak(bestDonorPwsm, mbrPeak.Apex.IndexedPeak.RetentionTime);
+
+                        if (peptideSpectralMatches == null || !peptideSpectralMatches.Any())
+                        {
+                            bestMbrMatches.Add(new MbrSpectralMatch(null, mbrPeak));
+                        }
+                        else
+                        {
+                            bestMbrMatches.Add(new MbrSpectralMatch(BestPsmForMbrPeak(peptideSpectralMatches),
+                                mbrPeak));
```

Comment:
Do you need to enclose these in a lock?

Reply:
bestMbrMatches is a ConcurrentBag, which means locks are unnecessary